livekit-agents/livekit/agents/pipeline/room_io.py

```diff
@@ -50,123 +50,338 @@ class RoomOutputOptions:
 
-        self._participant_ready = asyncio.Event()
         self._room.on("participant_connected", self._on_participant_connected)
+        self._room.on("participant_disconnected", self._on_participant_disconnected)
+        for participant in self._room.remote_participants.values():
+            self._on_participant_connected(participant)
 
-        # try to find participant
-        # TODO: support link and unlink participants
-        if self._expected_identity is not None:
-            participant = self._room.remote_participants.get(self._expected_identity)
-            if participant is not None:
-                self._link_participant(participant)
-        else:
-            for participant in self._room.remote_participants.values():
-                self._link_participant(participant)
-                if self._participant:
-                    break
+    @property
+    def audio(self) -> AsyncIterator[rtc.AudioFrame] | None:
+        if not self._audio_handle:
+            return None
+        return self._audio_handle.stream
 
-        self._tasks: set[asyncio.Task] = set()
+    @property
+    def video(self) -> AsyncIterator[rtc.VideoFrame] | None:
+        if not self._video_handle:
+            return None
+        return self._video_handle.stream
+
+    @property
+    def linked_participant(self) -> rtc.RemoteParticipant | None:
+        if not self._participant_connected.done():
+            return None
+        return self._participant_connected.result()
 
     async def start(self, agent: Optional["PipelineAgent"] = None) -> None:
-        await self.wait_for_participant()
+        participant = await self.wait_for_participant()
+        if self._participant_identity is None:
+            # link to the first connected participant if not set
+            self.set_participant(participant.identity)
+
         if not agent:
             return
 
         agent.input.audio = self.audio
         agent.input.video = self.video
         if self._options.forward_user_transcript:
-            # TODO: support multiple participants
             self._text_sink = RoomTranscriptEventSink(
-                room=self._room, participant=self._participant, is_stream=False
+                room=self._room, participant=participant, is_stream=False
             )
             agent.on("user_transcript_updated", self._on_user_transcript_updated)
 
-    @property
-    def audio(self) -> AsyncIterator[rtc.AudioFrame] | None:
-        if self._audio_stream is None:
-            return None
+    def set_participant(self, participant_identity: str | None) -> None:
+        """Switch audio and video streams to specified participant"""
+        if participant_identity is None:
+            self.unset_participant()
+            return
 
-        async def _read_stream():
-            async for event in self._audio_stream:
-                yield event.frame
+        # reset future if switching to a different participant
+        if (
```

Comment:
when a new participant is set.. should we automatically trigger a interrupt event? for PTT functionality, it seems like you'd want that?

Reply:
makes sense.. I think we should set up native RPCs.. but not for this PR.